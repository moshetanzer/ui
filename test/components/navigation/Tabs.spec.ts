import { describe, it, expect } from 'vitest'
import { UTabs } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Tabs', () => {
  it.each([
    ['with items', { props: { items: [{ label: 'Item 1' }, { label: 'Item 2' }] } }],
    ['with orientation', { props: { items: [{ label: 'Item 1' }, { label: 'Item 2' }], orientation: 'vertical' } }],
    ['with default slot', { slots: { default: () => 'Default slot' }, props: { items: [{ label: 'Item 1' }, { label: 'Item 2' }] } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UTabs.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UTabs)
    expect(html).toMatchSnapshot()
  })
})
