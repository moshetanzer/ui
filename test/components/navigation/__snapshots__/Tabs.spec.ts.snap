// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > renders with default slot correctly 1`] = `
"<div class="relative space-y-2">
  <div role="tablist" aria-orientation="horizontal" class="relative bg-gray-100 dark:bg-gray-800 rounded-lg p-1 w-full h-10 inline-grid items-center" style="grid-template-columns: repeat(2, minmax(0, 1fr));">
    <div class="absolute top-[4px] left-[4px] duration-200 ease-out focus:outline-none" style="top: 0px; left: 0px; width: 0px; height: 0px;">
      <div class="w-full h-full bg-white dark:bg-gray-900 rounded-md shadow-sm"></div>
    </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-900 dark:text-white" id="headlessui-tabs-tab-8" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected" type="button" aria-controls="headlessui-tabs-panel-10">Default slot</button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-500 dark:text-gray-400" id="headlessui-tabs-tab-9" role="tab" aria-selected="false" data-headlessui-state="" type="button" aria-controls="headlessui-tabs-panel-11">Default slot</button>
  </div>
  <div class="relative w-full">
    <div id="headlessui-tabs-panel-10" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-8"></div>
    <div id="headlessui-tabs-panel-11" role="tabpanel" hidden="" style="display: none;" data-headlessui-state="" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-9"></div>
  </div>
</div>"
`;

exports[`Tabs > renders with items correctly 1`] = `
"<div class="relative space-y-2">
  <div role="tablist" aria-orientation="horizontal" class="relative bg-gray-100 dark:bg-gray-800 rounded-lg p-1 w-full h-10 inline-grid items-center" style="grid-template-columns: repeat(2, minmax(0, 1fr));">
    <div class="absolute top-[4px] left-[4px] duration-200 ease-out focus:outline-none" style="top: 0px; left: 0px; width: 0px; height: 0px;">
      <div class="w-full h-full bg-white dark:bg-gray-900 rounded-md shadow-sm"></div>
    </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-900 dark:text-white" id="headlessui-tabs-tab-0" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected" type="button" aria-controls="headlessui-tabs-panel-2"><span class="truncate">Item 1</span></button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-500 dark:text-gray-400" id="headlessui-tabs-tab-1" role="tab" aria-selected="false" data-headlessui-state="" type="button" aria-controls="headlessui-tabs-panel-3"><span class="truncate">Item 2</span></button>
  </div>
  <div class="relative w-full">
    <div id="headlessui-tabs-panel-2" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-0"></div>
    <div id="headlessui-tabs-panel-3" role="tabpanel" hidden="" style="display: none;" data-headlessui-state="" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-1"></div>
  </div>
</div>"
`;

exports[`Tabs > renders with orientation correctly 1`] = `
"<div class="relative space-y-2">
  <div role="tablist" aria-orientation="vertical" class="relative bg-gray-100 dark:bg-gray-800 rounded-lg p-1 w-full">
    <div class="absolute top-[4px] left-[4px] duration-200 ease-out focus:outline-none" style="top: 0px; left: 0px; width: 0px; height: 0px;">
      <div class="w-full h-full bg-white dark:bg-gray-900 rounded-md shadow-sm"></div>
    </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-900 dark:text-white" id="headlessui-tabs-tab-4" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected" type="button" aria-controls="headlessui-tabs-panel-6"><span class="truncate">Item 1</span></button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-500 dark:text-gray-400" id="headlessui-tabs-tab-5" role="tab" aria-selected="false" data-headlessui-state="" type="button" aria-controls="headlessui-tabs-panel-7"><span class="truncate">Item 2</span></button>
  </div>
  <div class="relative w-full">
    <div id="headlessui-tabs-panel-6" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-4"></div>
    <div id="headlessui-tabs-panel-7" role="tabpanel" hidden="" style="display: none;" data-headlessui-state="" class="focus:outline-none" aria-labelledby="headlessui-tabs-tab-5"></div>
  </div>
</div>"
`;
